### PR TITLE
Fix autocomplete and highlighting for new integer types

### PR DESCRIPTION
**Pull request by nathanfranke**
Also, changed the order of these types to make more sense

**Before**:
image

**After**:
image

### CODE REVIEW

File: modules/gdscript/gdscript_editor.cpp [hunk 1]
```diff
@@ -2172,7 +2172,7 @@ static void _find_identifiers(const GDScriptCompletionContext &p_context, bool p
 	}
 
 	static const char *_type_names[Variant::VARIANT_MAX] = {
-		"null", "bool", "int", "float", "String", "Vector2", "Rect2", "Vector3", "Transform2D", "Plane", "Quat", "AABB", "Basis", "Transform",
```
Comment (akien-mga) [inline modules/gdscript/gdscript_editor.cpp]: I think the order should be the same as in the `Variant::TYPE` enum. I see that `"StringName"` is missing too, could you add it?

File: editor/plugins/script_text_editor.cpp [hunk 2]
```diff
@@ -292,9 +292,12 @@ void ScriptTextEditor::_set_theme_for_script() {
 	const Color basetype_color = colors_cache.basetype_color;
 	text_edit->add_keyword_color("String", basetype_color);
 	text_edit->add_keyword_color("Vector2", basetype_color);
+	text_edit->add_keyword_color("Vector3", basetype_color);
```
Comment (akien-mga) [inline editor/plugins/script_text_editor.cpp]: While at it, please reorder like in `variant.h` / `_type_names`.

**Comment by nathanwfranke** [comment]
@akien-mga I also added `StringName` and `Callable` to the GDScript keywords since they are now supported. Let me know if I should remove them

**Comment by akien-mga** [comment]
Thanks!

File: editor/plugins/script_text_editor.cpp [hunk 3]
```diff
@@ -292,24 +292,29 @@ void ScriptTextEditor::_set_theme_for_script() {
...
 	text_edit->add_keyword_color("String", basetype_color);
 	text_edit->add_keyword_color("Vector2", basetype_color);
+	text_edit->add_keyword_color("Vector2i", basetype_color);
 	text_edit->add_keyword_color("Rect2", basetype_color);
-	text_edit->add_keyword_color("Transform2D", basetype_color);
+	text_edit->add_keyword_color("Rect2i", basetype_color);
 	text_edit->add_keyword_color("Vector3", basetype_color);
+	text_edit->add_keyword_color("Vector3i", basetype_color);
+	text_edit->add_keyword_color("Transform2D", basetype_color);
+	text_edit->add_keyword_color("Plane", basetype_color);
```
Comment (aaronfranke) [inline editor/plugins/script_text_editor.cpp:301]: You can do multi-line reviews, Nathan.

Comment (nathanfranke) [reply editor/plugins/script_text_editor.cpp:301]: Ok

**Pull request merged**